packages/vuetify/src/labs/VNumberInput/__tests__/VNumberInput.spec.cy.tsx

```diff
@@ -8,6 +8,38 @@ import { VForm } from '@/components/VForm'
 import { ref } from 'vue'
 
 describe('VNumberInput', () => {
+  it('should prevent NaN from arbitrary input', () => {
+    const scenarios = [
+      { text: '---', expected: '-' }, // "-" is only allowed once
+      { text: '1-', expected: '1' }, // "-" is only at the start
+      { text: '.', expected: '' }, // "." isn't allowed at the start
+      { text: '1...0', expected: '1.0' }, // "." is only allowed once
+      { text: '123.45.67', expected: '123.4567' }, // "." is only allowed once
```

Comment:
I guess this doesn't handle the european reversal of `,` and `.`?

Reply:
At this stage, number-input only supports basic numeric input. Trying to get the basic function right.

In future, might consider integrating it with masks or a new `stringMode` prop to be able to support some advanced things

Reply:
Or we can accept `locale` prop and extract separators ([example](https://github.com/J-Sek/vuetify/blob/feature-19898/packages/vuetify/src/util/helpers.ts#L805))